packages/adapters/storage/src/postgres.rs

```diff
@@ -976,6 +965,146 @@ impl Postgres {
 
         Ok(())
     }
+
+    pub(crate) async fn _prune_entries_older_than(
+        &self,
+        date: DateTime<Utc>,
+    ) -> Result<services::state_pruner::port::Pruned> {
+        let response = sqlx::query!(
+            r#"
+            WITH
```

Comment:
Since this query has multiple mutating queries, you should wrap the whole thing with a `BEGIN` and `END` so that, if it fails in the middle, you're not left with a half-deleted state.  Or use a sqlx transaction.

Seeing as you depend on the ids of the deleted entries not being lost, it is important that the whole statement passes or nothing.

Reply:
used `sqlx transaction`